src/memory_region.rs

```diff
@@ -386,7 +411,9 @@ impl<'a> AlignedMemoryMapping<'a> {
     #[inline]
     pub fn find_region(&self, vm_addr: u64) -> Option<(usize, &MemoryRegion)> {
         let index = vm_addr.wrapping_shr(ebpf::VIRTUAL_ADDRESS_BITS as u32) as usize;
-        if (1..self.common.regions.len()).contains(&index) {
+        if index < self.common.regions.len()
+            || (index == 0 && !self.common.config.allow_memory_region_zero)
+        {
```

Comment:
Since we only want to include index `0` when `allow_memory_region_zero` is `true`, shouldn't this instead be written like this?

```suggestion
        if index < self.common.regions.len()
            && (index != 0 || self.common.config.allow_memory_region_zero)
        {
```

Otherwise you end up with this logic:

```rust
let allow_memory_region_zero = false;
let index = 0;

if 0 < 5 || (0 == 0 && !false) // if true || (true && true)
```

In fact, you never even hit the second logic gate at all if index is `0`.

Reply:
https://github.com/anza-xyz/sbpf/pull/96